src/main/java/core/basesyntax/UpdateStorageImpl.java

```diff
@@ -0,0 +1,51 @@
+package core.basesyntax;
+
+import java.io.IOException;
+import java.time.LocalDate;
+import java.util.List;
+import java.util.TreeMap;
+
+public class UpdateStorageImpl implements UpdateStorage {
+
+    public void parseFileToStorage(String fileName) throws IOException {
+        List<List<String>> newData = new ReadFromFile().getNewData(fileName);
+        for (List<String> raw : newData) {
+            putLineToStorage(raw);
+        }
+    }
+
+    public void putLineToStorage(List<String> raw) {
```

Comment:
try to split this method into smaller parts, it will be easier to read code this way

Reply:
done